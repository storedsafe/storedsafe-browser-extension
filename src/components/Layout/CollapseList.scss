@import '../../variables';

.collapse-list-item {
  background-color: $color-bg;

  &.padded {
    padding: $spacing;
  }

  &:nth-child(even) {
    background-color: $color-bg-alt;
  }

  &:last-child {
    border-bottom: 1px solid $color-bg;
  }
}
