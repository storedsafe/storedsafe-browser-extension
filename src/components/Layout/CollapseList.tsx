import * as React from 'react';
import { CollapseBox } from './';
import './CollapseList.scss';

export interface CollapseListProps {
  startCollapsed?: boolean;
  title: React.ReactNode;
  items: React.ReactNode[];
  padded?: boolean;
}

export const CollapseList: React.FC<CollapseListProps> = ({
  startCollapsed,
  title,
  items,
  padded,
}: CollapseListProps) => {
  const children = items.map((item, index) => (
    <div key={index} className={`collapse-list-item${padded ? ' padded' : ''}`}>
      {item}
    </div>
  ));
  return (
    <div className="collapse-list">
      <CollapseBox padded={false} startCollapsed={startCollapsed} title={title}>
        {children}
      </CollapseBox>
    </div>
  );
}

CollapseList.defaultProps = {
  startCollapsed: true,
  padded: true,
};
