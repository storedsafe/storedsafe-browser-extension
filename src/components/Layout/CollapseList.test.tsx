import * as React from 'react';
import { shallow } from 'enzyme';
import { CollapseList } from './CollapseList';

test('<CollapseList />', () => {
  const items = [
  <p key={0}>Foo</p>,
  <p key={1}>Bar</p>,
  <p key={2}>Zot</p>
  ];
  const wrapper = shallow(<CollapseList items={items} title="Title" />);
  expect(wrapper).toMatchSnapshot();
});

test('<CollapseList collapsed=false />', () => {
  const items = [
  <p key={0}>Foo</p>,
  <p key={1}>Bar</p>,
  <p key={2}>Zot</p>
  ];
  const wrapper = shallow(<CollapseList
    items={items}
    title="Title"
    startCollapsed={false}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('<CollapseList collapsed=true />', () => {
  const items = [
  <p key={0}>Foo</p>,
  <p key={1}>Bar</p>,
  <p key={2}>Zot</p>
  ];
  const wrapper = shallow(<CollapseList
    items={items}
    title="Title"
    startCollapsed={true}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('<CollapseList padded=false />', () => {
  const items = [
  <p key={0}>Foo</p>,
  <p key={1}>Bar</p>,
  <p key={2}>Zot</p>
  ];
  const wrapper = shallow(<CollapseList
    items={items}
    title="Title"
    padded={false}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('<CollapseList padded=true />', () => {
  const items = [
  <p key={0}>Foo</p>,
  <p key={1}>Bar</p>,
  <p key={2}>Zot</p>
  ];
  const wrapper = shallow(<CollapseList
    items={items}
    title="Title"
    padded={true}
  />);
  expect(wrapper).toMatchSnapshot();
});
