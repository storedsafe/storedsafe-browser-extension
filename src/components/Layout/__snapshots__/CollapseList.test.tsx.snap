// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollapseList /> 1`] = `ShallowWrapper {}`;

exports[`<CollapseList collapsed=false /> 1`] = `ShallowWrapper {}`;

exports[`<CollapseList collapsed=true /> 1`] = `ShallowWrapper {}`;

exports[`<CollapseList padded=false /> 1`] = `ShallowWrapper {}`;

exports[`<CollapseList padded=true /> 1`] = `ShallowWrapper {}`;
